src/components/KTableView/KTableView.vue

```diff
@@ -798,15 +799,23 @@ const getHeaderClasses = (column: TableViewHeader<ColumnKey>, index: number): Re
 const onHeaderClick = (column: TableViewHeader<ColumnKey>) => {
   if (column.sortable && column.key !== TableViewHeaderKeys.BULK_ACTIONS && column.key !== TableViewHeaderKeys.ACTIONS) {
     let newSortColumnOrder: SortColumnOrder = 'asc'
-    if (column.key === sortColumnKey.value && sortColumnOrder.value === 'asc') {
+    let newSortColumnKey: ColumnKey | '' = column.key
+    if (newSortColumnKey === sortColumnKey.value && sortColumnOrder.value === 'asc') {
       newSortColumnOrder = 'desc'
     }
+    // Reset sorting column when clicking on same column while in descending order
+    // this mimics a three-state toggle: 'asc' -> 'desc' -> 'none'
+    if (newSortColumnKey === sortColumnKey.value && sortColumnOrder.value === 'desc') {
+      newSortColumnKey = ''
+      newSortColumnOrder = DEFAULT_INITIAL_SORT_ORDER
+    }
+
     emit('sort', {
       prevKey: sortColumnKey.value,
-      sortColumnKey: column.key,
+      sortColumnKey: newSortColumnKey,
       sortColumnOrder: newSortColumnOrder,
     })
-    sortClickHandler(column)
+    sortClickHandler(newSortColumnKey)
```

Comment:
It’s not very clear to me why we need a separate `sortClickHandler` inside the header click handler, since it’s essentially serving as the sort handler itself. But since this wasn’t introduced in this PR, I’m fine with leaving it as is for now. I do think it would be better to extract a pure function that determines the next sort state based on the previous one. That way, the click handler could be much simpler. The current code looks good overall, but I’d suggest refactoring it later.